addons/pkg-npm/jaxrs/src/main/java/org/commonjava/indy/pkg/npm/jaxrs/PackageContentAccessHandler.java

```diff
@@ -0,0 +1,425 @@
+
+public class PackageContentAccessHandler
+                extends ContentAccessHandler
+{
+    public static final String CROSS = "-";
+
+    public static final String PATH_SPLITER = "/";
+
+    public static final String TGZ = ".tgz";
+
+    public static final String PACAGE_TGZ = "package" + TGZ;
+
+    @Override
+    public Response doHead( final String packageType, final String type, final String name, final String packageName,
+                            final Boolean cacheOnly, final String baseUri, final HttpServletRequest request,
+                            EventMetadata eventMetadata, final Consumer<Response.ResponseBuilder> builderModifier )
+    {
+        return doHead( packageType, type, name, packageName, null, null, cacheOnly, baseUri, request, eventMetadata,
+                       builderModifier );
+    }
+
+    @Override
+    public Response doHead( final String packageType, final String type, final String name, final String packageName,
+                            final Boolean cacheOnly, final String baseUri, final HttpServletRequest request,
+                            EventMetadata eventMetadata )
+    {
+        return doHead( packageType, type, name, packageName, null, null, cacheOnly, baseUri, request, eventMetadata,
+                       null );
+    }
+
+    public Response doHead( final String packageType, final String type, final String name, final String packageName,
+                            final String version, final String tarball, final Boolean cacheOnly, final String baseUri,
+                            final HttpServletRequest request, EventMetadata eventMetadata )
+    {
+        return doHead( packageType, type, name, packageName, version, tarball, cacheOnly, baseUri, request,
+                       eventMetadata, null );
+    }
+
+    public Response doHead( final String packageType, final String type, final String name, final String packageName,
+                            final String version, final String tarball, final Boolean cacheOnly, final String baseUri,
+                            final HttpServletRequest request, EventMetadata eventMetadata,
+                            final Consumer<Response.ResponseBuilder> builderModifier )
+    {
+        if ( !PackageTypes.contains( packageType ) )
+        {
+            Response.ResponseBuilder builder = Response.status( 400 );
+            if ( builderModifier != null )
+            {
+                builderModifier.accept( builder );
+            }
+            return builder.build();
+        }
+
+        final StoreType st = StoreType.get( type );
+        final StoreKey sk = new StoreKey( packageType, st, name );
+
+        eventMetadata = eventMetadata.set( ContentManager.ENTRY_POINT_STORE, sk );
+
+        final AcceptInfo acceptInfo = jaxRsRequestHelper.findAccept( request, ApplicationContent.text_html );
+
+        Response response = null;
+
+        if ( packageName == null || packageName.trim().length() < 1 )
```

Comment:
It seems like you have a REST API class that maps specific NPM paths to different methods, which is great because it controls the use cases tightly in a situation where they are well known, and avoids a lot of bound checking for that. But having all of them call a modified version of ContentAccessHandler that takes all the fields and then has to do a bunch of bounds checking for nulls/empty values seems...complicated.

Would it make more sense to implement the /{package} -> /{package}/package.json mappings in the appropriate REST API methods directly, and then just use the original ContentAccessHandler? I'm not sure if it would mean having duplicate path translations in your REST methods, but you could always implement private methods in the REST API class to help do those mappings, and then just call them wherever that mapping was needed.

I think you could take advantage of the separate REST methods you were able to achieve, and avoid needing this new access handler class completely...but maybe I'm missing something?